ethcore/src/factory.rs

```diff
@@ -15,14 +15,44 @@
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
 use trie::TrieFactory;
-use evm::Factory as EvmFactory;
 use account_db::Factory as AccountFactory;
+use evm::{Factory as EvmFactory, VMType};
+use vm::{Vm, ActionParams, Schedule};
+use wasm::WasmInterpreter;
+
+const WASM_MAGIC_NUMBER: &'static [u8; 4] = b"\0asm";
+
+/// Virtual machine factory
+#[derive(Default, Clone)]
+pub struct VmFactory {
+	evm: EvmFactory,
+}
+
+impl VmFactory {
+	pub fn create(&self, params: &ActionParams, schedule: &Schedule) -> Box<Vm> {
+		if schedule.wasm.is_some() && params.code.as_ref().map_or(false, |code| code.len() > 4 && &code[0..4] == WASM_MAGIC_NUMBER) {
```

Comment:
Does WASM work with `gas > u64::max_value()`? I think I saw couple of `low_u64` calls below.

Reply:
It will fail later in the vm itself:

https://github.com/paritytech/parity/blob/master/ethcore/wasm/src/lib.rs#L85